hypersync-client/src/from_arrow.rs

```diff
@@ -1,307 +1,305 @@
+use std::fmt::Debug;
+
 use arrayvec::ArrayVec;
-use polars_arrow::array::{
-    BinaryArray, BinaryViewArray, BooleanArray, StaticArray, UInt64Array, UInt8Array, Utf8Array,
-    Utf8ViewArray,
+use arrow::array::{
+    Array, BinaryArray, BooleanArray, RecordBatch, StringArray, UInt64Array, UInt8Array,
 };
+use hypersync_format::{TransactionStatus, TransactionType, UInt};
 
-use crate::{
-    simple_types::{Block, Log, Trace, Transaction},
-    ArrowBatch,
-};
+use crate::simple_types::{Block, Log, Trace, Transaction};
 
-/// Used to do ArrowBatch-Native Rust type conversions while consuming the input value.
+/// Used to convert arrow `RecordBatch` into a vector of structs of the appropriate type like
+/// Log, Transaction, Block etc.
 pub trait FromArrow: Sized {
-    /// Converts to the Vector type from the ArrowBatch type.
-    /// B is the type of binary array used to downcast since we use both binary views and
-    /// binary arrays
-    fn from_arrow_bin_array<B: List<Item = [u8]> + 'static, U: List<Item = str> + 'static>(
-        batch: &ArrowBatch,
-    ) -> Vec<Self>;
-
-    /// Default implementation that uses the binary array type.
-    /// IPC is written with regular binary arrays.
-    /// Named 'from_arrow' for backwards compatibility.
-    fn from_arrow(batch: &ArrowBatch) -> Vec<Self> {
-        Self::from_arrow_bin_array::<BinaryArray<i32>, Utf8Array<i32>>(batch)
-    }
-
-    /// An additional method that uses the binary view array type.
-    /// This is to be able to reuse the trait server side where Binary Views are
-    /// used instead.
-    fn from_arrow_bin_view_array(batch: &ArrowBatch) -> Vec<Self> {
-        Self::from_arrow_bin_array::<BinaryViewArray, Utf8ViewArray>(batch)
-    }
+    /// Convert given record batch to a vector of structs
+    fn from_arrow(batch: &RecordBatch) -> Vec<Self>;
```

Comment:
Nice, we don't need this trait any more if we're only dealing with binary/utf8

Reply:
Removed this trait